connector/connect/common/src/main/protobuf/spark/connect/relations.proto

```diff
@@ -598,3 +595,10 @@ message ToSchema {
   // The Sever side will update the dataframe with this schema.
   DataType schema = 2;
 }
+
+// Explicitly sort the underlying plan to make the ordering deterministic for Sample.
+// Note: this message is only used to randomly splits DataFrame with the provided weights.
+message DeterministicOrder {
```

Comment:
maybe also document that when there is no explicit ordering then the input plan is cached?

Reply:
DeterministicOrder just used for explicit ordering. It means the underlying dataframe doesn't guarantee the ordering of rows.